pyartifactory/objects.py

```diff
@@ -545,3 +571,148 @@ def delete(self, permission_name: str) -> None:
         self.get(permission_name)
         self._delete(f"api/{self._uri}/{permission_name}")
         logging.debug(f"Repository {permission_name} successfully deleted")
+
+
+class ArtifactoryArtifact(ArtifactoryObject):
+    def __init__(self, artifactory: AuthModel) -> None:
+        super(ArtifactoryArtifact, self).__init__(artifactory)
+
+    def deploy(
+        self, local_file_location: str, artifact_path: str
+    ) -> ArtifactPropertiesResponse:
+        """
+        :param artifact_path: Path to file in Artifactory
+        :param local_file_location: Location of the file to deploy
+        """
+        artifact_path = artifact_path.lstrip("/")
+        local_filename = artifact_path.split("/")[-1]
+        try:
+            with open(local_file_location, "rb") as f:
+                form = encoder.MultipartEncoder(
+                    {
+                        "documents": (local_filename, f, "application/octet-stream"),
+                        "composite": "NONE",
+                    }
+                )
+                headers = {"Prefer": "respond-async", "Content-Type": form.content_type}
+                self._put(f"{artifact_path}", headers=headers, data=form)
+                logging.info(f"Artifact {local_filename} successfully deployed")
+                return self.properties(artifact_path)
+        except ArtifactDeployException:
+            logging.error(f"Cannot deploy artifact {local_filename}")
+            raise
+
+    def download(self, artifact_path: str) -> str:

```

Comment:
Also the previous comment makes me think: maybe we should take an optional parameter as a file path (or a Path object) and we write to it, that way users can control where the file lands (and we still provide the default behavior as a fallback, for ease of use).